draft-cpbs-pseudorandom-ctls.md

```diff
@@ -190,15 +178,23 @@ In general, a malicious peer can still produce desired ciphertext with probabili
 
 Representing plaintext alerts (i.e. CTLSPlaintext messages with `content_type = alert(21)`) requires additional steps, because Alert fragments have little entropy.
 
-A standard TLS Alert fragment is always 2 bytes long.  In Pseudorandom cTLS, senders MUST prepend at least 16 random bytes to any plaintext Alert fragment.  Enciphering and deciphering then proceed identically to other CTLSPlaintext messages.  The recipient MUST remove these bytes before passing the CTLSPlaintext to the cTLS implementation.
-
-Servers SHOULD expand any Alert message following the ClientHello to the same size as their usual ServerHello, and SHOULD send additional random TCP segments or datagrams to match the sizes of subsequent components of their ordinary success response.  Otherwise, an adversary could use probing to learn the allowed lengths of ClientHellos and the fraction of ciphertexts that decipher to valid ClientHellos.
+A standard TLS Alert fragment is always 2 bytes long.  In Pseudorandom cTLS, senders MUST append at least 16 random bytes to any plaintext Alert fragment and increase `CTLSPlaintext.length` accordingly.  Enciphering and deciphering then proceed identically to other CTLSPlaintext messages.  The recipient MUST remove these bytes before passing the CTLSPlaintext to the cTLS implementation.
 
 > QUESTION: Are there client-issued Alerts in response to malformed ServerHello?
 
 # Operational Considerations
 
-Pseudorandom cTLS can interfere with the use of multiple profiles on a single server.  To use Pseudorandom cTLS with multiple profiles, servers must use the same TSPRP key and the same lengths of `connection_id`.
+## Multiple profiles and key rotation
```

Comment:
It sounds like this section is just clarifying what is already the current state of affairs: If multiple profiles may be in use, the server will have to test each in order to process a ClientHello. Is this right?

Reply:
Yes, this section doesn't invent anything; it just repurposes `profile_id` as a MAC.  However, it does have normative recommendations because:

1. You need the `profile_id` to be long enough that you don't get a false-positive match by chance when deciphering under the wrong profile.
2. Oddly, the two profiles can have the same `profile_id`, since it's only being used here as a MAC.
3. This is what allows mixing Pseudorandom and non-Pseudorandom cTLS on the same port.
4. Even with trial deciphering, connection IDs still don't work quite right, so their use is restricted.